_includes/com-ingest/com-ingest-extra-kafka-avro-field-syntax.md

```diff
@@ -0,0 +1,74 @@
+### Kafka Avro data type syntax
+
+Map Avro field data types and property key-value pairs to determine the FeatureBase record data type
+
+#### BOOL
+
+| Kafka Avro fields | Description |
+|---|---|
+| `{"name": "bool_bool", "type": "boolean"}` | FeatureBase Bool from Avro Boolean
+
+#### DECIMAL
+
+| Kafka Avro fields | Description |
+|---|---|
+| `{"name": "decimal_float", "type": "float", "fieldType": "decimal", "scale": 2}` | FeatureBase Decimal from Avro Float |
+
+#### ID
+
+| Kafka Avro fields | Description |
+|---|---|
+| `{"name": "id_long", "type": "long", "mutex": true, "fieldType": "id"}` | FeatureBase ID from Avro Long |
+| `{"name": "id_int", "type": "int", "mutex": true, "fieldType": "id"}` | FeatureBase ID from Avro int |
+
+#### IDSET
+
+| Kafka Avro fields | Description |
+|---|---|
+| `{"name": "idset_int", "type": "int", "fieldType": "id"}` | FeatureBase IDSET from Avro Int |
+| `{"name": "idset_intarray", "type": {"type": "array", "items": "int"}}` | FeatureBase IDSET from Avro Int Array |
+
+#### IDSETQ
+
+| Kafka Avro schema | Kafka Avro fields | Description |
+|---|---|---|
+| N/A | `{"name": "idsetq_int", "type": "int", "fieldType": "id", "quantum": "YMD"}` | FeatureBase IDSETQ from Avro Int |
+| `{"name": "recordtime_bytes", "type": "bytes", "fieldType": "recordTime", "layout": "2006-01-02 15:04:05", "unit": "s"},` | `{"name": "idsetq_intarray", "type": "array", "items": {"type": "int", "quantum": "YMD"}}`| FeatureBase IDSETQ from Avro Int Array.<br/>`RecordTime` field is also required in the Avro Schema. |
+
+#### INT
+
+| Kafka Avro fields | Description |
+|---|---|
+| `{"name": "int_int", "type": "int", "fieldType": "int"}` | FeatureBase Int from Avro Int |
+
+#### Strings
+
+| Kafka Avro fields | Description | Additional |
+|---|---|---|
+| `{"name": "string_string", "type": "string", "mutex": true }`| FeatureBase String from Avro String |  |
+| `{"name": "string_bytes", "type": "bytes" , "mutex": true }` | FeatureBase String from Avro Bytes |  |
+| `{"name": "string_enum", "type": "enum"}` | FeatureBase String from Avro Enum |  |
+| `{"name": "string_string", "type": ["string", "null"], "mutex": true }` | Optional String | [Ignore missing fields](#ignore-missing-fields) |
+| `{"name": "stringset_stringarray", "type": [{"type": "array", "items": "string"}, "null"]}` | Optional Array of Strings | [Ignore missing fields](#ignore-missing-fields) |
+
```

Comment:
Something I should've made clear up-front is that I'm not an expert in SQL or databases :) I turn what experts write into *possibly* clearer content :)

With this in mind, I think you're saying that all Avro fields can have a union with Null.

What's the context though? Because there *is* a separate include file which discusses the use of Null values to skip fields:

`/_includes/com-ingest/com-ingest-extra-kafka-avro-fields.md`

This could probably benefit from a better title such as : `/com-ingest-extra-kafka-avro-ignore-fields.md`

Reply:
Yep there is a distinction between missing values and passing a value of null through avro. The behavior could be different but not entirely sure. But, I'm good moving forward with how it is now. We can add more later.

Reply:
sounds like a plan :)